src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -118,6 +118,11 @@ public function createRequest(Request $request, string $path)
     public function checkRequestPath(Request $request, string $path)
     {
         if ('/' !== $path[0]) {
+            // Shortcut if request has already been matched before
+            if ($request->attributes->has('_route')) {
+                return $path === $request->attributes->get('_route');
+            }
```

Comment:
You might have already considered this, but what about instead caching this in the URL matcher? This means other calls to `matchRequest()` also benefit from this optimization. My initial feeling is that this logic is an "implementation detail" of the URL matcher, rather than the Security component.

Reply:
I have and it would be doable of course. Having it here just solves that one problem for potentially a lot of other users, not just for us :)

Reply:
I agree with Wouter, the `_route` attribute is an implementation detail we should not rely on here.

Reply:
What exactly is the implementation detail? The fact that we store `_route` as a request attribute? Or the fact that it is named `_route`? I mean, I get your point but I'd argue that the entire `HttpUtils::checkRequestPath()` only makes sense to be used in the Symfony fullstack framework kind of context because it already relies on the fact that the parameter is named `_route`. I mean, look at the entire method now, it's meant to check if the url matcher returns a matching `_route` parameter.

Reply:
Well, the UrlMatcher is guaranteed to return `_route` on matches. However, the URL matcher is not the one storing anything in request attributes (that's done by the RoutingListener of HttpKernel).

So to me, doing that optimization inside the routing component looks wrong to me. The component that guarantes that `$request->attributes->get('_route')` contains the matching route after matching is done is the HttpKernel one.

However, I would not treat that route attribute as an internal implementation detail of HttpKernel. It has been documented since years and lots of packages are relying on it.

Reply:
I vote for keeping the current implementation of this PR.

Reply:
We already rely on that parameter [10 lines below](https://github.com/symfony/symfony/pull/51986/files#diff-09ea53eab5f9bcb734da5c64182bfa774b1e3e63a209f2c1e444dd83ce7fd70aR134) so this makes sense as is to me also.